### DISCUSSION TITLE
init --prompt* asks for the whole prompt, is this intended functionality?

### DISCUSSION THREAD

**Discussion by frantathefranta** [twpayne/chezmoi#3225] [Q&A] [answered]
I'm not sure if this is the correct functionality or if my setup is just incorrect. 
When I try to run `chezmoi init --promptString email="fb@franta.us" --promptBool hasGUI="true"`, I still get prompted in the shell.
It's only when I pass the "prompt description", that it doesn't prompt me and fills in the data automatically. Example:
`chezmoi init --promptString "Email address"="fb@franta.us" --promptBool "Does this machine have a GUI [y/n]"=true`
 
This is my config:
```
login-node% vi .local/share/chezmoi/.chezmoi.toml.tmpl
# vim: set ft=toml:
{{- $email := promptStringOnce . "email" "Email address" -}}
{{- $hasGUI := promptBoolOnce . "hasGUI" "Does this machine have a GUI [y/n]" -}}

[data]
    email = {{ $email | quote }}
    hasGUI = {{ $hasGUI | quote }}
```

I'd appreciate any advice, obviously it works, but makes the command rather wordy. I feel like I'm missing something rather than this being intended.

**Comment by nandalopes** [comment]
`init --promptString` uses the prompt, not the path in the map, to know what to fill.

See [chezmoi init command](https://www.chezmoi.io/reference/commands/init/#-promptstring-pairs) and [promptStringOnce template function](https://www.chezmoi.io/reference/templates/init-functions/promptStringOnce/)

**Reply by frantathefranta** [reply]
So I am using it correctly? I'm still a little confused.

**Reply by twpayne** [reply, marked as answer]
> So I am using it correctly?

No. The `--promptBool` flags are for testing your config file template with `chezmoi execute-template` so you don't have to re-enter the answers each time.

If you want to set up a machine automatically, either use template logic in your config file template, or use an environment variable.

For example:

```
{{ $email := "" }}
{{ if eq .chezmoi.hostname "work-machine" }}
{{   $email = "franta@work.com" }}
{{ else if eq .chezmoi.hostname "home-machine" }}
{{   $email = "fb@home.org" }}
{{ else }}
{{   $email = promptStringOnce . "email" "Email address" }}
{{ end }}
```

or

```
{{ $email := env "EMAIL" }}
{{ if not $email }}
{{   $email = promptStringOnce . "email" "Email address" }}
{{ end }}
```

**Reply by frantathefranta** [reply]
Ah, this makes sense. Thank you for your explanation!

**Comment by nandalopes** [comment]
If you're using the same defaults, why don't you add a default value directly on your template?

**Reply by frantathefranta** [reply]
My goal is to make this work in an Ansible playbook, so having defaults is a good idea, but I still need a way to set what I want.
